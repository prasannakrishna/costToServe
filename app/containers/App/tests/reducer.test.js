/**
 * Copyright Â© 2018, JDA Software Group, Inc. ALL RIGHTS RESERVED.
 * <p>
 * This software is the confidential information of JDA Software, Inc., and is licensed
 * as restricted rights software. The use,reproduction, or disclosure of this software
 * is subject to restrictions set forth in your license agreement with JDA.
 */
import { fromJS } from 'immutable';
import appReducer from '../reducer';
import {
  loadRepos,
  reposLoaded,
  repoLoadingError,
  logout,
  getAlertsCount,
  alertsCountLoaded,
  alertsCountError,
  getMetaData,
  metaDataLoaded,
  metaDataLoadingError,
  getBusinessProcesses,
  businessProcessesLoaded,
  businessProcessesLoadingError,
} from '../actions';

const initialState = fromJS({
  loading: false,
  error: false,
  currentUser: false,
  userData: {
    repositories: false,
  },
  metaData: false,
  businessProcesses: false,
  buildNumber: false,
});

describe('appReducer', () => {
  it('returns the initial state', () => {
    expect(appReducer(undefined, {})).toEqual(initialState);
  });
  it('should handles loadRepos', () => {
    expect(appReducer(initialState, loadRepos([{}]))).toMatchSnapshot();
  });
  it('should handles logout', () => {
    expect(appReducer(initialState, logout([{}]))).toMatchSnapshot();
  });
  it('should handles reposLoaded', () => {
    expect(appReducer(initialState, reposLoaded([{}]))).toMatchSnapshot();
  });
  it('should handles getAlertsCount', () => {
    expect(appReducer(initialState, getAlertsCount([{}]))).toMatchSnapshot();
  });
  it('should handles alertsCountLoaded', () => {
    expect(appReducer(initialState, alertsCountLoaded([{}]))).toMatchSnapshot();
  });
  it('should handles getMetaData', () => {
    expect(appReducer(initialState, getMetaData([{}]))).toMatchSnapshot();
  });
  it('should handles alertsCountError', () => {
    expect(appReducer(initialState, alertsCountError([{}]))).toMatchSnapshot();
  });
  it('should handles metaDataLoaded', () => {
    expect(appReducer(initialState, metaDataLoaded([{}]))).toMatchSnapshot();
  });
  it('should handles metaDataLoadingError', () => {
    expect(appReducer(initialState, metaDataLoadingError([{}]))).toMatchSnapshot();
  });
  it('should handles getBusinessProcesses', () => {
    expect(appReducer(initialState, getBusinessProcesses([{}]))).toMatchSnapshot();
  });
  it('should handles businessProcessesLoaded', () => {
    expect(appReducer(initialState, businessProcessesLoaded([{}]))).toMatchSnapshot();
  });
  it('should handles businessProcessesLoadingError', () => {
    expect(appReducer(initialState, businessProcessesLoadingError([{}]))).toMatchSnapshot();
  });
  it('should handles repoLoadingError', () => {
    expect(appReducer(initialState, repoLoadingError([{}]))).toMatchSnapshot();
  });
});
