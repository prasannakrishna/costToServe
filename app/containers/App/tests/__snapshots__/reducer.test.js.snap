// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appReducer should handles alertsCountError 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "currentUser": false,
  "userData": Immutable.Map {
    "repositories": false,
  },
  "metaData": false,
  "businessProcesses": false,
  "buildNumber": false,
}
`;

exports[`appReducer should handles alertsCountLoaded 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "currentUser": false,
  "userData": Immutable.Map {
    "repositories": false,
  },
  "metaData": false,
  "businessProcesses": false,
  "buildNumber": false,
}
`;

exports[`appReducer should handles businessProcessesLoaded 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "currentUser": false,
  "userData": Immutable.Map {
    "repositories": false,
  },
  "metaData": false,
  "businessProcesses": Array [
    Object {},
  ],
  "buildNumber": false,
}
`;

exports[`appReducer should handles businessProcessesLoadingError 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "currentUser": false,
  "userData": Immutable.Map {
    "repositories": false,
  },
  "metaData": false,
  "businessProcesses": false,
  "buildNumber": false,
}
`;

exports[`appReducer should handles getAlertsCount 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "currentUser": false,
  "userData": Immutable.Map {
    "repositories": false,
  },
  "metaData": false,
  "businessProcesses": false,
  "buildNumber": false,
}
`;

exports[`appReducer should handles getBusinessProcesses 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "currentUser": false,
  "userData": Immutable.Map {
    "repositories": false,
  },
  "metaData": false,
  "businessProcesses": false,
  "buildNumber": false,
}
`;

exports[`appReducer should handles getMetaData 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "currentUser": false,
  "userData": Immutable.Map {
    "repositories": false,
  },
  "metaData": false,
  "businessProcesses": false,
  "buildNumber": false,
}
`;

exports[`appReducer should handles loadRepos 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
  "currentUser": false,
  "userData": Immutable.Map {
    "repositories": false,
  },
  "metaData": false,
  "businessProcesses": false,
  "buildNumber": false,
}
`;

exports[`appReducer should handles logout 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "currentUser": false,
  "userData": Immutable.Map {
    "repositories": false,
  },
  "metaData": false,
  "businessProcesses": false,
  "buildNumber": false,
}
`;

exports[`appReducer should handles metaDataLoaded 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "currentUser": false,
  "userData": Immutable.Map {
    "repositories": false,
  },
  "metaData": Array [
    Object {},
  ],
  "businessProcesses": false,
  "buildNumber": false,
}
`;

exports[`appReducer should handles metaDataLoadingError 1`] = `
Immutable.Map {
  "loading": false,
  "error": Array [
    Object {},
  ],
  "currentUser": false,
  "userData": Immutable.Map {
    "repositories": false,
  },
  "metaData": false,
  "businessProcesses": false,
  "buildNumber": false,
}
`;

exports[`appReducer should handles repoLoadingError 1`] = `
Immutable.Map {
  "loading": false,
  "error": Array [
    Object {},
  ],
  "currentUser": false,
  "userData": Immutable.Map {
    "repositories": false,
  },
  "metaData": false,
  "businessProcesses": false,
  "buildNumber": false,
}
`;

exports[`appReducer should handles reposLoaded 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "currentUser": undefined,
  "userData": Immutable.Map {
    "repositories": Array [
      Object {},
    ],
  },
  "metaData": false,
  "businessProcesses": false,
  "buildNumber": false,
}
`;
